apps/backend/src/action/logic/check-tool-has-broken.ts

```diff
@@ -1,27 +1,85 @@
-import { EventSource } from "~/models/event";
-import { Event } from "~/action/event";
+import { ResultAsync } from "neverthrow";
+import { pipe } from "fp-ts/lib/function";
+import { match } from "ts-pattern";
+import * as Array from "fp-ts/lib/Array";
+import * as E from "fp-ts/Either";
 import { BrokenToolCommand } from "~/action/command";
-import { ResultAsync, errAsync } from "neverthrow";
+import {
+  Event,
+  isBrokenToolHasBeenPlacedEvent,
+  isBrokenToolHasBeenRemovedEvent,
+} from "~/action/event";
+import type { EventSource } from "~/models/event";
+import { Tool } from "~/models/tool";
+import { always, error, identity } from "~/utils";
+
+const EventSourceReadError = error("EventSourceReadError");
+
+type EventSourceReadError = ReturnType<typeof EventSourceReadError>;
+
+export type CheckToolHasBrokenError = EventSourceReadError;
 
 export interface CheckToolHasBroken {
   (repository: EventSource<Event>, command: BrokenToolCommand): ResultAsync<
     boolean,
```

Comment:
@JohnsonMao could you help me change `boolean` to `BrokenToolCommand`, thank you!

Reply:
@kayac-chang Sure, I'd be happy to help.
Just to confirm, you want to change the `boolean` to `BrokenToolCommand`, correct? Not `Tool[]`?

Because if we change it to `BrokenToolCommand`, this function will only return `BrokenToolCommand` or `EventSourceReadError` without performing any other actions.

Please let me know if my understanding is correct.

Reply:
yes, if this check passing will just return the original command.

Reply:
If the check fails, should the function also return the original command or false or ResultError?

Reply:
if the check fails, the function should return `ResultError`
which will fit the type requirement.
```ts
type CheckToolsHasBroken = {
  (repository: EventSource<Event>, command: BrokenToolCommand): ResultAsync<
    BrokenToolCommand,
    CheckToolHasBrokenError
  >;
}
```